vktrace/src/vktrace_extensions/vktracevulkan/vkreplay/vkreplay_vkreplay.cpp

```diff
@@ -1566,8 +1566,22 @@ VkResult vkReplay::manually_replay_vkWaitForFences(packet_vkWaitForFences* pPack
     {
         (*(pFence + i)) = m_objMapper.remap_fences((*(pPacket->pFences + i)));
     }
-    replayResult = m_vkFuncs.real_vkWaitForFences(remappedDevice, pPacket->fenceCount, pFence, pPacket->waitAll, pPacket->timeout);
-    VKTRACE_DELETE(pFence);
+	if (pPacket->result == VK_SUCCESS)
+	{
+		replayResult = m_vkFuncs.real_vkWaitForFences(remappedDevice, pPacket->fenceCount, pFence, pPacket->waitAll, UINT64_MAX);// mean as long as possible
+	}
```

Comment:
After reviewing a bit more, I think the existing solution is the best way to most closely replicate the original applications behavior. If the original app only called vkWaitForFences() once and got a success, this will best ensure that we can get success by calling vkWaitForFences() only once. 

If the original app got a TIMEOUT, then we'll replay the call once with the minimum amount of time (0) in hopes of getting a TIMEOUT. Note that the only way to truly guarantee a timeout is to skip the call completely and just return a TIMEOUT, but then we're not replaying a call that the application made so it's probably not a great solution. In an actual app, we'd expect them to make this call again, so they'll likely see SUCCESS on an upcoming call.

If the call returned an error, then it really doesn't matter what we set the timeout value to, because we're just taking a chance anyway.

Based on the above logic, I think this multi-conditional approach is reasonable.